tests/unit/amo/reducers/test_recommendations.js

```diff
@@ -0,0 +1,84 @@
+import reducer, {
+  abortFetchRecommendations,
+  fetchRecommendations,
+  getRecommendationsByGuid,
+  initialState,
+  loadRecommendations,
+} from 'amo/reducers/recommendations';
+import { createInternalAddon } from 'core/reducers/addons';
+import { createStubErrorHandler } from 'tests/unit/helpers';
+import { fakeAddon } from 'tests/unit/amo/helpers';
+
+
+describe(__filename, () => {
+  it('initializes properly', () => {
+    const state = reducer(undefined, {});
+    expect(state).toEqual(initialState);
+  });
+
+  it('ignores unrelated actions', () => {
+    const state = reducer(initialState, { type: 'UNRELATED_ACTION' });
+    expect(state).toEqual(initialState);
+  });
+
+  it('sets the loading flag when fetching recommendations', () => {
+    const guid = 'some-guid';
+    const state = reducer(undefined, fetchRecommendations({
+      errorHandlerId: createStubErrorHandler().id,
+      guid,
+      recommended: true,
+    }));
+
+    expect(state.byGuid[guid].loading).toEqual(true);
+    expect(state.byGuid[guid].addons).toEqual(undefined);
```

Comment:
According to the flow def, there is always an `addon` array that can be empty.

Reply:
I updated that so it can be null, and updated this test.